api/app/invitationletter/api.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime
+import flask_restful as restful
+from sqlalchemy.exc import IntegrityError
+from app.utils.auth import verify_token
+from flask import g, request
+from app.invitationletter.models import InvitationTemplate
+from app.registration.models import Offer
+from app.invitationletter.mixins import InvitationMixin
+from app.invitationletter.models import InvitationLetterRequest
+from app.invitationletter.generator import generate
+from app.users.models import AppUser, Country
+from app import db, LOGGER
+from app.utils import errors
+from app.utils.auth import auth_required
+
+
+def invitation_info(invitation_request):
+    return {
+        'invitation_letter_request_id': invitation_request.id,
+    }
+
+
+class InvitationLetterAPI(InvitationMixin, restful.Resource):
+
+    @auth_required
+    def post(self):
+        args = self.req_parser.parse_args()
+        registration_id = args['registration_id']
+        event_id = args['event_id']
+        work_address = args['work_address']
+        addressed_to = args['addressed_to']
+        residential_address = args['residential_address']
+        passport_name = args['passport_name']
+        passport_no = args['passport_no']
+        passport_issued_by = args['passport_issued_by']
+        to_date = datetime.strptime((args['to_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        from_date = datetime.strptime((args['from_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        user_id = verify_token(request.headers.get('Authorization'))['id']
+
+        invitation_letter_request = InvitationLetterRequest(
+            registration_id=registration_id,
+            event_id=event_id,
+            work_address=work_address,
+            addressed_to=addressed_to,
+            residential_address=residential_address,
+            passport_name=passport_name,
+            passport_no=passport_no,
+            passport_issued_by=passport_issued_by,
+            invitation_letter_sent_at=datetime.now().strftime("%Y-%m-%d"),
+            to_date=to_date,
+            from_date=from_date,
+        )
+
+        offer = db.session.query(Offer).filter(
+            Offer.user_id == user_id).first()
+
+        if not offer:
+            return errors.OFFER_NOT_FOUND
+
+        global invitation_template
+
+        if offer.accommodation_award and offer.travel_award:
```

Comment:
Please also check accepted_accommodation_award and accepted_travel_award, so the condition should be something like: 
```if offer.accommodation_award and offer.accepted_accommodation_award and offer.travel_award and offer.accepted_travel_award:```
Same applies for the below two